src/lib.rs

```diff
@@ -10,13 +10,12 @@
 #![forbid(unsafe_code)]
 #![warn(missing_docs, rust_2018_idioms, unused_qualifications)]
 
-mod register;
+pub mod register;
```

Comment:
Since you've re-exported `Register` you don't need this. All of the types defined wherein are now public via the toplevel.

```suggestion
mod register;
```

Reply:
Understood, thank you. I was still mistakenly importing via register::Register so it wasn't compiling locally, but it seems to work fine with a direct Register import.